crates/apollo-compiler/src/database/hir.rs

```diff
@@ -996,6 +1010,12 @@ impl Field {
     pub fn loc(&self) -> &HirNodeLocation {
         &self.loc
     }
+
+    /// returns true if this is an introspection field (i.e. it's [`Self::name()`] is one of __type, or __schema).
+    pub fn is_introspection(&self) -> bool {
+        let field_name = self.name();
+        field_name == "__type" || field_name == "__schema"
```

Comment:
also `__typename` is a possible introspection field
```suggestion
        field_name == "__type" || field_name == "__schema" || field_name == "__typename"
```

Reply:
oh I see.. When I was writing this, I wasn't sure if `__typename` counted since it's not mentioned at https://graphql.org/learn/introspection/. But I'll defer that to you.